Sources/Private/CoreAnimation/ValueProviderStore.swift

```diff
@@ -36,7 +36,9 @@ final class ValueProviderStore {
       properties. Supported properties are: \(supportedProperties.joined(separator: ", ")).
       """)
 
-    valueProviders.append((keypath: keypath, valueProvider: valueProvider))
+    orderedKeyPaths.remove(keypath)
+    orderedKeyPaths.insert(keypath, at: orderedKeyPaths.count)
```

Comment:
This is just an "append" right? Can we use a built-in method like https://developer.apple.com/documentation/foundation/nsmutableorderedset/1408009-add?

```suggestion
    orderedKeyPaths.add(keypath)
```